flow/project.py

```diff
@@ -675,14 +678,22 @@ def __call__(self, func):
             The decorated function.
 
         """
+        if not any(
+            func == op_func for _, op_func in self._project._OPERATION_FUNCTIONS
+        ):
+            raise FlowProjectDefinitionError(
+                f"Cannot add function '{func}' to group without making the function an "
+                f"operation. Add @MyProjectClass.operation below group decorator."
+            )
+
         if hasattr(func, "_flow_groups"):
```

Comment:
Since `FlowProject.operation` creates a `_flow_groups` attribute we should just check for the existence of that and error if it is not found. Although, if it passes the operation test, then the attribute should always exist.

Reply:
Are you saying that if `_OPERATION_FUNCTIONS` is populated then we must have also populated `_flow_groups`? If so, I agree. What would you like to see here now, then?

Reply:
We could remove the reference to the project that this adds and makes the class slightly more complicated and just check for the existence of `_flow_groups` and error if not found.